### ISSUE TITLE
feat: auto refresh by watching each directory in $PATH

### ISSUE THREAD

**Issue by WieeRd** [WieeRd/auto-lsp.nvim#4]
`auto-lsp.nvim` cannot detect servers that are newly installed while Neovim is running.
This is because the result of availability check of each server/filetype is cached.
Auto refreshing upon `{ "FocusGained", "TermLeave" }` has already been implemented as a PoC,
but watching $PATH would be far more accurate and require less manual intervention.

**Comment by WieeRd** [comment]
[`:h uv_fs_event_t`](https://neovim.io/doc/user/luvref.html#uv_fs_event_t)

**Comment by xyven1** [comment]
Just a note: to correctly detect new binaries when using direnv (with nix specifically, but I'm sure in other situatinos as well), you would need to watch the variable `$PATH` itself for changes, as direnv will modify path to include the executables directly, not using by adding executables to paths already in `$PATH`.

**Comment by WieeRd** [comment]
Yeah, that's a bit of a headache since I am not aware of any method capable of detecting `vim.env` changes other than periodically scanning. However, unless you use `direnv.vim` to directly integrate direnv with Neovim, I don't think the value of `vim.env.PATH` will change while Neovim is running. The environment variables are only captured on startup from the parent shell, right?

**Comment by WieeRd** [comment]
Also I'm kind of reluctant on this feature as a whole since the band-aid `{ "FocusGained", "TermLeave" }` solution has worked out pretty well. The plugin naturally refreshes itself after returning from the external / embedded terminal you've used to install new packages, and you can also just tab out and come back to trigger a refresh.

**Comment by xyven1** [comment]
Yeah, I agree honestly. I feel like I will more often then not end up closing neovim to install a lsp server (why not, when it takes significantly less than a second for neovim to open?).
